{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4dcc2159",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "22c3c9fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_clinical = 'DXSUM_12Jan2026.csv'\n",
    "file_proteomics = 'RBM/adni_proteomicsv2/adni_plasma_qc_multiplex_11Nov2010.csv'\n",
    "file_expression = 'ADNI_Gene_Expression_Profile/ADNI_Gene_Expression_Profile.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fc712463",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   -> Found 3788 patients in Clinical Master File\n",
      "   -> Found 566 patients with Inflammatory Markers\n"
     ]
    }
   ],
   "source": [
    "# LOADING CLINICAL DATA\n",
    "df_clin = pd.read_csv(file_clinical, low_memory=False)\n",
    "rids_clinical = set(df_clin['RID'].astype(int).unique())\n",
    "print(f\"   -> Found {len(rids_clinical)} patients in Clinical Master File\")\n",
    "\n",
    "# LOADING PROTEOMICS\n",
    "df_prot = pd.read_csv(file_proteomics)\n",
    "rids_prot = set(df_prot['RID'].astype(int).unique())\n",
    "print(f\"   -> Found {len(rids_prot)} patients with Inflammatory Markers\")\n",
    "\n",
    "# LOADING TRANSCRIPTOMICS \n",
    "# Read first 10 rows with NO header to see raw structure\n",
    "df_check = pd.read_csv(file_expression, header=None, nrows=10, low_memory=False)\n",
    "\n",
    "rids_rna = set()\n",
    "found_row = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fac000e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   -> Detected Patient IDs in Row Index 2\n",
      "   -> Found 744 patients with Gene Expression Data\n"
     ]
    }
   ],
   "source": [
    "# Loop through the first 10 rows to find the one containing IDs\n",
    "for i in range(10):\n",
    "    row_values = df_check.iloc[i].values.astype(str)\n",
    "    # looking for the \"_S_\" pattern\n",
    "    # We check if at least 3 columns have that pattern to be sure\n",
    "    matches = [x for x in row_values if '_S_' in x]\n",
    "    \n",
    "    if len(matches) > 5:  # Found it!\n",
    "        print(f\"   -> Detected Patient IDs in Row Index {i}\")\n",
    "        found_row = True\n",
    "        \n",
    "        # Process this specific row\n",
    "        for ptid in matches:\n",
    "            try:\n",
    "                # Convert '116_S_1249' -> 1249\n",
    "                rid_part = int(ptid.split('_')[-1])\n",
    "                rids_rna.add(rid_part)\n",
    "            except:\n",
    "                continue\n",
    "        break\n",
    "\n",
    "if not found_row:\n",
    "    print(\"   ! ERROR: Could not locate a row with '_S_' IDs in the first 10 lines.\")\n",
    "else:\n",
    "    print(f\"   -> Found {len(rids_rna)} patients with Gene Expression Data\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4f2f3ef4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FINAL RESULT: 170 PATIENTS HAVE ALL 3 DATA TYPES\n"
     ]
    }
   ],
   "source": [
    "# FINDING INTERSECTION\n",
    "# Find patients who have ALL THREE\n",
    "golden_cohort = rids_clinical.intersection(rids_prot).intersection(rids_rna)\n",
    "count = len(golden_cohort)\n",
    "\n",
    "print(f\"FINAL RESULT: {count} PATIENTS HAVE ALL 3 DATA TYPES\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cb321f5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
